SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -118,6 +135,57 @@ void ASpatialDebugger::BeginPlay()
 			SpatialToggleDebugger();
 		}
 	}
+	else
+	{
+		if (NetDriver->LoadBalanceStrategy != nullptr)
```

Comment:
In a multiserver scenario, I think all servers will calculate this array, in addition to all the non-auth servers (and clients) receiving the OnRep (though I see the servers ignore the OnRep).  Would it be cleaner for only the auth server to calculate it, and just allow the replication to send it out to the rest?

Reply:
I think it's just all non-auth servers calculating currently, but yea agree it should just be authoritative. I tried just doing it on the auth server by checking the authority on the GSM but that seems not to work - is this how you would do it? At first glance I can't find other relevant logic in SpatialDebugger for the authoritative worker

Reply:
Hmm that should work. The master/all-powerful/auth server should be denoted by having authority over the GSM 🤔

Reply:
So interesting when BeginPlay is called on singletons (or this singleton at least), the GSM HasAuthority returns false meaning it wasn't immediate to knowing worker was the authoritative one. As a simple solution, I moved this to the `OnAuthorityGained` function inherited from AActor and now it works 😄